server/src/main/java/org/elasticsearch/repositories/VerifyNodeRepositoryCoordinationAction.java

```diff
@@ -0,0 +1,159 @@
+import org.elasticsearch.transport.TransportException;
+import org.elasticsearch.transport.TransportResponseHandler;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class VerifyNodeRepositoryCoordinationAction extends ActionType<VerifyNodeRepositoryCoordinationAction.Response> {
+
+    public static final VerifyNodeRepositoryCoordinationAction INSTANCE = new VerifyNodeRepositoryCoordinationAction();
+    public static final String NAME = "internal:admin/repository/verify/coordinate";
+
+    private VerifyNodeRepositoryCoordinationAction() {
+        super(NAME);
+    }
+
+    public static class Response extends ActionResponse {
+
+        final List<DiscoveryNode> nodes;
+
+        public Response(List<DiscoveryNode> nodes) {
+            this.nodes = nodes;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            TransportAction.localOnly();
+        }
+    }
+
+    public static class LocalAction extends TransportAction<Request, Response> {
+
+        private final TransportService transportService;
+        private final ClusterService clusterService;
+        private final NodeClient client;
+
+        @Inject
+        public LocalAction(
+            ActionFilters actionFilters,
+            TransportService transportService,
+            ClusterService clusterService,
+            NodeClient client
+        ) {
+            super(NAME, actionFilters, transportService.getTaskManager());
+            this.transportService = transportService;
+            this.clusterService = clusterService;
+            this.client = client;
+        }
+
+        @Override
+        protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
+            final DiscoveryNodes discoNodes = clusterService.state().nodes();
+            final DiscoveryNode localNode = discoNodes.getLocalNode();
+
+            final Collection<DiscoveryNode> masterAndDataNodes = discoNodes.getMasterAndDataNodes().values();
+            final List<DiscoveryNode> nodes = new ArrayList<>();
+            for (DiscoveryNode node : masterAndDataNodes) {
+                if (RepositoriesService.isDedicatedVotingOnlyNode(node.getRoles()) == false) {
+                    nodes.add(node);
+                }
+            }
+            final CopyOnWriteArrayList<VerificationFailure> errors = new CopyOnWriteArrayList<>();
+            final AtomicInteger counter = new AtomicInteger(nodes.size());
+            for (final DiscoveryNode node : nodes) {
+                if (node.equals(localNode)) {
+                    client.executeLocally(VerifyNodeRepositoryAction.INSTANCE, request, new ActionListener<>() {
```

Comment:
Oh wait it's called via the client here. This whole branch isn't necessary tho, we can send transport requests to the local node just fine these days.